src/herder/Herder.h

```diff
@@ -116,8 +116,12 @@ class Herder
                                   SCPQuorumSet const& qset) = 0;
     virtual bool recvTxSet(Hash const& hash, TxSetFrame const& txset) = 0;
     // We are learning about a new transaction.
+    virtual TransactionQueue::AddResult
+    recvTransaction(TransactionFrameBasePtr tx, TransactionResult& txRes) = 0;
+#ifdef BUILD_TESTS
```

Comment:
General thought: test-only code in the public interface is typically a bad smell, as it both adds maintenance burden to the non-test implementers, as well as gets around the purpose of the tests (which normally should cover the public interface used by the real, non-test class clients).

For example, here it seems like we just use this method about 10 times in the tests, would adding a dummy output `txRes` for these tests work (instead of re-using the old method)?

Reply:
I agree with the general sentiment regarding the test-only code. Ideally, we shouldn't have them, since they potentially give rise to unit test logic diverging from actual (production) code, and it's harder to maintain (as you mentioned).
However, 1. we already have them in our current code base (this is not to give excuse for writing more of those) and 2. doing it in this PR saves a ton of boilerplate code change: 
The `TransactionFrameBase` class and its many implementations `TransactionFrame`, `FeeBumpTransactionFrame`, `Fuzzer*TransactionFrame` are pervasive in all the unit-test code, and many of them are currently accessing the `TransactionResult` via member variable, those callers would need to passed in external `TransactionResult` to the new interfaces. 
Although the `recvTransaction` method here is only called a couple dozen times, which makes good argument for not using test method here, I would lean towards having them since the main objective is to adapt tests in the bigger scope to the new interfaces without changing boilerplate code.

Reply:
In that case, I agree with you both. I've made the change to remove the test-only `recvTransaction`.

Reply:
This is fixed.